frontend/src/Views/MainPage/MainPage.tsx

```diff
@@ -7,54 +7,82 @@ import MainPageHeader from '@Commons/MainPageHeader/MainPageHeader';
 import FormModule from '@Commons/Forms/FormModule/FormModule.tsx';
-import * as inputModules from '@Commons/Forms/InputModules/index';
 import InputModuleInterface from 'src/Interface/InputModuleinterface';
 import FarmDetailsInterface from 'src/Interface/FarmDetailsInterface';
+import * as InputModules from '@Commons/Forms/InputModules/index';
 import { StyledMain, StyledMainContainer } from './MainPage.styles';
 
 // The sequence of sections to show up on the main page
 // This is the skeleton for the Berries workflow
 // Uncomment sections as they are implemented to have them instantiated ;)
-const mockBerriesWorkflow = [
-  { InputModuleID: 'FarmInformation' },
-  { InputModuleID: 'FieldsAndSoil' },
-  { InputModuleID: 'ManureAndCompost' },
-  { InputModuleID: 'Calculate' },
-  { InputModuleID: 'Summary' },
+const mockBerriesWorkflow: InputModuleInterface[] = [
+  InputModules.FarmInformation,
+  InputModules.FieldsAndSoil,
+  InputModules.ManureAndCompost,
+  InputModules.Calculate,
+  InputModules.Summary,
 ];
 
 // Initial Values for calculation, some defaults are being used
 const initialFarmDetails: FarmDetailsInterface = {
-  Year: '2024',
+  Year: '',
   FarmName: '',
   FarmRegion: '',
   HasBerries: true,
 };
 
 const MainPage: React.FC = () => {
   const [farmDetails, setFarmDetails] = useState(initialFarmDetails);
+  const [formStates, setFormStates] = useState(mockBerriesWorkflow);
 
+  /**
+   * @summary   Pass this handler to children who need to update InputModule states
+   * @desc      A State handler that will update the current form section states,
+   *            allowing you to expand/collapse form sections.
+   *            ** In the future, also update the ProgressBar status **
+   * @param     moduleID: string => An InputModule ID to be searched and updated
+   * @param     nextModuleID: string => A second InputModule ID to be searched and updated
+   */
+  const handleFormState = (moduleID: string, nexModuleID?: string) => {
+    const updatedStates = formStates.map((module: InputModuleInterface) => {
+      if (module.id === moduleID || module.id === nexModuleID) {
+        return {
+          ...module,
+          enable: !module.enable,
+        };
+      }
+      return module;
+    });
+    setFormStates(updatedStates);
+  };
+
+  /**
+   * @summary   Handler for updating the Main Data of the Calculator.
+   * @desc      This updates the Main Data objet being built, 'farmDetails'.
+   *            Since after an update, the form should take you to the next section,
+   *            it calls the form state handler to update states automatically <3
+   * @param     newDetails => A new 'FarmDetailsInterface' object with new data from
+   *            from a form section.
+   * */
   const updateFarmDetails = (newDetails: FarmDetailsInterface) => {
     setFarmDetails(newDetails);
+    handleFormState(InputModules.FarmInformation.id, InputModules.FieldsAndSoil.id);
```

Comment:
We could include going through summary as well, as it's already basically implemented, so that we can test entirely that it's going through next sections, and it's not using hardcoded values like this.